Configurations/10-main.conf

```diff
@@ -2080,4 +2100,43 @@ my %targets = (
         pointer_size     => "64",
     },
 
+#### CMake
+    "CMake-common" => {
+        inherit_from     => [ 'BASE_CMake' ],
+        template         => 1,
+        cppflags         => combine(threads("-D_MT")),
+        lib_cppflags     => "",
+        ex_libs          => "",
+        thread_scheme    => "(unknown)",
+        dso_scheme       => "",
+        shared_target    => "cmake-shared",
+        shared_cppflags  => "",
+        shared_ldflag    => "",
+
+        perl_platform    => 'CMake',
+    },
+    "CMake" => {
+        inherit_from     => [ "CMake-common" ],
+        sys_id           => "CMake",
+        CFLAGS           => "",
+        cflags           => "",
+        AS               => sub { cmake_info()->{AS} },
+        ASFLAGS          => sub { cmake_info()->{ASFLAGS} },
+        asoutflag        => sub { cmake_info()->{asoutflag} },
+        asflags          => sub { cmake_info()->{asflags} },
+        ld_resp_delim    => "\n",
+        asm_arch         => 'x86_64',
+        perlasm_scheme   => "auto",
+        bn_ops           => "SIXTY_FOUR_BIT RC4_CHAR",
+        # As for uplink_arch. Applink makes it possible to use
+        # .dll compiled with one compiler with application compiled with
+        # another compiler. It really shouldn't be needed when embedding OpenSSL
+        # into a project using CMake.
+        # Perhaps this should be configurable via the config script?
```

Comment:
It *is* configurable:

https://github.com/openssl/openssl/blob/d79bb5316e1318bd776d6b2d6723a36778e07f9d/INSTALL.md?plain=1#L1021-L1023